src/app/screens/executeBrc20Transaction/index.tsx

```diff
@@ -0,0 +1,104 @@
+import useWalletSelector from '@hooks/useWalletSelector';
+import { ExecuteBrc20TransferState } from '@utils/brc20';
+import { ExecuteTransferProgressCodes, useBrc20TransferExecute } from '@secretkeylabs/xverse-core';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { getBtcTxStatusUrl } from '@utils/helper';
+import { ConfirmationStatus } from '@components/loadingTransactionStatus/circularSvgAnimation';
+import { useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { LoadingTransactionStatus } from '@components/loadingTransactionStatus';
+
+const AMOUNT_OF_STEPS = Object.keys(ExecuteTransferProgressCodes).length + 1;
+const PERCENTAGE_PER_STEP = 1 / AMOUNT_OF_STEPS;
+
+export function ExecuteBrc20Transaction() {
+  const { t } = useTranslation('translation', { keyPrefix: 'EXECUTE_BRC20' });
+  const { selectedAccount, btcAddress, network, seedPhrase } = useWalletSelector();
+  const navigate = useNavigate();
+  const { recipientAddress, estimateFeesParams }: ExecuteBrc20TransferState = useLocation().state;
+
+  const { progress, complete, executeTransfer, transferTransactionId, errorCode } =
+    useBrc20TransferExecute({
+      ...estimateFeesParams,
+      seedPhrase,
+      accountIndex: selectedAccount?.id ?? 0,
+      changeAddress: btcAddress,
+      recipientAddress,
+      network: network.type,
+    });
+
+  const [loadingPercentage, setLoadingPercentage] = useState(0);
+
+  useEffect(() => {
+    if (complete || errorCode) {
+      return;
+    }
+    executeTransfer();
+  }, [executeTransfer, complete, errorCode]);
```

Comment:
I think we only want to execute executeTransfer once on load. Executing it more than once would result in an error somewhere along the way. This might be enough?

```suggestion
  useEffect(() => {
    executeTransfer();
  }, []);
```